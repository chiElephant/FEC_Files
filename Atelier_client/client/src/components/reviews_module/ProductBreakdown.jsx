import React, { useState, useEffect } from 'react';

function ProductBreakdown({ characteristics }) {
  const [charsDiv, setCharsDiv] = useState(<div />);

  useEffect(() => {
    createCharsDiv();
  }, [characteristics]);

  function createCharsDiv() {
    const newCharDiv = [];

    const poorAndGreat = [
      <div id="poor" key="poor">
        Poor
      </div>,
      <div id="great" key="great">
        Great
      </div>,
    ];

    const bigAndSmall = [
      <div id="too-small" key="small">
        Too Small
      </div>,
      <div id="perfect" key="perfect">
        Perfect
      </div>,
      <div id="too-big" key="big">
        Too Big
      </div>,
    ];

    for (let char in characteristics) {
      if (characteristics[char] !== null) {
        const key = characteristics[char].id;
        const value = Math.floor(characteristics[char].value);
        let element;

        if (char === 'Comfort' || char === 'Quality') {
          element = poorAndGreat;
        } else {
          element = bigAndSmall;
        }
        newCharDiv.push(
          <div className="characteristic" key={key}>
            <div className="char-name">{`${char}`}</div>
            <div className="char-meter">
              <i
                className="char-icon"
                className="fak fa-triangle-solid fa-flip-vertical"
                style={{ marginLeft: value * 20 + '%' }}
              ></i>
            </div>
            <div className="breakdown-descrip">{element}</div>
          </div>
        );
      }
    }

    setCharsDiv((charsDiv) => newCharDiv);
  }

  return (
    <div>
      <h3 hidden>Product Breakdown</h3>
      {charsDiv}
    </div>
  );
}

export default ProductBreakdown;
