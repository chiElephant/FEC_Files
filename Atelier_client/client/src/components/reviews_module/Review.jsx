import React, { useState, useEffect } from 'react';
import { helpers } from './helper_functions/review.js';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid, regular, light, thin } from '@fortawesome/fontawesome-svg-core/import.macro';

function Review({ review }) {

  const [photosDiv, setPhotosDiv] = useState(<div />);
  const [helpfulness, setHelpfulness] = useState(0);
  const [recommend, setRecommend] = useState(false);
  const [response, setResponse] = useState(null);
  const [review_id, setReview_id] = useState(0);
  const [username, setUsername] = useState('');
  const [stars, setStars] = useState(<div />);
  const [summary, setSummary] = useState('');
  const [photos, setPhotos] = useState([]);
  const [rating, setRating] = useState(0);
  const [body, setBody] = useState('');
  const [date, setDate] = useState('');

  useEffect(() => {
    handleReviewData(review)
  }, [review])

  useEffect(() => {
    if (review) {
      async function handleDate () {
        const convertedDate = await helpers.convertDate(review.date);
        setDate(convertedDate);
      }
      handleDate()

    }
  }, [review]);

  useEffect(() => {
    async function handlePhotos () {
      const photos = await helpers.handlePhotos(review.photos);
      setPhotosDiv(photos);
    }
    handlePhotos();
  }, [review]);

  useEffect(() => {
    handleRatings(review.rating);
    handleRecommend(review.recommend);
    handleResponse(review.response);
  }, [review]);

  useEffect(() => {
    handleHelpfulness(review.helpfulness);
  }, [review]);

  function handleReviewData(review) {
    setReview_id(review.review_id);
    setUsername(review.reviewer_name.concat(','));
    setSummary(review.summary);
    setRating(review.rating);
    setPhotos(review.photos);
    setBody(review.body);
  }

  function handleRatings(rating) {
    let starRating = []
    if (rating !== 0) {
      for (let i = 1; i <= rating; i++) {
        starRating.push(
          <FontAwesomeIcon
            key={`${i}-solid`}
            className='star'
            icon={solid('star-sharp')}
          />
        )
      }
      for (let i = rating; i < 5; i++) {
        starRating.push(
          <FontAwesomeIcon
            key={`${i}-regular`}
            className='star'
            icon={light('star-sharp')}
          />
        )
      }
    }
    setStars(starRating);
  }

  function handleRecommend (recommend) {
    let recDiv = [];

    if (recommend) {
      recDiv.push(
        <div key='recommend' id='user-recommend'>
          <FontAwesomeIcon
            key='1'
            icon={regular('check')}
          />
          <span id='user-rec'>
            {'I recommend this product'}
          </span>
        </div>
      );
      setRecommend(recDiv);
    }
  }

  function handleResponse (response) {
    let responseDiv = [];

    if (response) {
      responseDiv.push(
        <div key='response' id='user-response'>
         <div className='response'>
          {'Response:'}
         </div>
         <div className='user-response'>
          {`${response}`}
         </div>
        </div>
      )
      setResponse(responseDiv)
    }
  }

  function handleHelpfulness (helpful) {
    let helpfulDiv = [];

    helpfulDiv.push(
      <div id='helpful-text' key='helpful'>
        <div>
          {'Helpful?'}
        </div>
        <div
        className='helpful-yes'
        onClick={() => markHelpful()}
        >
          {'Yes '}
        </div>
        <div>
          {`(${helpful})`}
        </div>
        <div>
          <FontAwesomeIcon
            icon={thin('pipe')}
          />
        </div>
        <div>
          {'Report'}
        </div>
      </div>
    )
    setHelpfulness(helpfulDiv);
  }

  function markHelpful() {

    const addHelpful = helpfulness + 1;
    handleHelpfulness(addHelpful);

    helpers.markHelpful(review_id);
  }

  return (
    <div id="review">
      <div id='review-top-bar'>
        <div id='stars'>{stars}</div>
        <div id='date'>{date}</div>
        <div id='username'>{username}</div>
      </div>
      <div id='summary'>{summary}</div>
      <div id='body'>{body}</div>
      <div id='photos'>{photosDiv}</div>
      <div id='recommend'>{recommend}</div>
      <div id={response !== null ? 'recommend' : ''}>{response}</div>
      <div id='helpfulness'>{helpfulness}</div>
    </div>
  );
}

export default Review;
