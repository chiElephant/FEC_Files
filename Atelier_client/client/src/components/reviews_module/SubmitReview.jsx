import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid, regular } from '@fortawesome/fontawesome-svg-core/import.macro';

function SubmitReview({
  showReviewModal,
  setShowReviewModal,
  submitReviewForm,
  productName,
  product_id,
  chars,
}) {
  const [productChars, setProductChars] = useState({});
  const [addPhotoDiv, setAddPhotoDiv] = useState(<div />);
  const [charsDiv, setCharsDiv] = useState(<div />);
  const [recommend, setRecommend] = useState(false);
  const [summary, setSummary] = useState("");
  const [rating, setRating] = useState(0);
  const [photos, setPhotos] = useState([]);
  const [email, setEmail] = useState("");
  const [body, setBody] = useState("");
  const [name, setName] = useState("");

  useEffect(() => {
    createCharsDiv();
  }, [chars]);

  const handleChange = (cb, value) => {
    cb(`${value}`);
  };

  const handleSubmit = () => {
    setShowReviewModal(false);
    axios
      .post('/reviews', {
        product_id,
        rating,
        summary,
        body,
        recommend,
        name,
        email,
        photos,
        characteristics: productChars,
      })
      .then((response) => {
        console.log(response);
      })
      .catch((error) => {
        console.log('post error: ', error);
      });
  };

  const charsKey = {
    Size: {
      1: 'A size too small',
      2: '1/2 size too small',
      3: 'Perfect',
      4: '1/2 size too big',
      5: 'A size too wide',
    },
    Width: {
      1: 'Too narrow',
      2: 'Slightly narrow',
      3: 'Perfect',
      4: 'Slightly wide',
      5: 'Too wide',
    },
    Comfort: {
      1: 'Uncomfortable',
      2: 'Slightly uncomfortable',
      3: 'Ok',
      4: 'Comfortable',
      5: 'Perfect',
    },
    Quality: {
      1: 'Poor',
      2: 'Below Average',
      3: 'What I expected',
      4: 'Pretty great',
      5: 'Perfect',
    },
    Length: {
      1: 'Runs short',
      2: 'Runs slightly short',
      3: 'Perfect',
      4: 'Runs slightly long',
      5: 'Runs long',
    },
    Fit: {
      1: 'Runs tight',
      2: 'Runs slightly tight',
      3: 'Perfect',
      4: 'Runs slightly long',
      5: 'Runs long',
    },
  };

  const handleClick = (cb, value) => {
    cb(value);
  };

  const handlePhotos = () => {
    let files = document.querySelector('#photo-input').files;
    let fileURLs = photos;

    if (files.length <= 5) {
      for (let i = 0; i < files.length; i++) {
        const file = files[i];

        // if (!file.type.startsWith('image/')) {
        //   continue;
        // }

        const img = document.createElement('img');
        img.file = file;
        img.width = 80;

        images.appendChild(img);

        const reader = new FileReader();
        reader.onload = (e) => {
          img.src = e.target.result;
          fileURLs.push(e.target.result);
          console.log(fileURLs)
          setPhotos(fileURLs);
        };
        reader.readAsDataURL(file);
      }
    }
  };

  const handleCharacteristics = (char, id, value) => {
    let chars = productChars;
    chars[`${id}`] = value;
    console.log(productChars)
    console.log(chars)
    setProductChars(chars);
    handleCharSelection(char, id, value);
  };

  const handleCharSelection = (char, id, value) => {
    const currentDisplay = document.getElementById(id);
    const newDisplay = charsKey[char][value];
    currentDisplay.textContent = newDisplay;
  };

  const createCharsDiv = () => {
    let div = [];

    for (let char in chars) {
      div.push(
        <div key={chars[char].id}>
          <span id={chars[char].id}>None Selected</span>
          <br />
          <label>
            {`${char}`}
            <input
              className={chars[char].id}
              type="radio"
              name={`${char}`}
              value="1"
              onClick={() =>
                handleCharacteristics(char, chars[char].id, 1)
              }
            />
            <input
              className={chars[char].id}
              type="radio"
              name={`${char}`}
              value="2"
              onClick={() =>
                handleCharacteristics(char, chars[char].id, 2)
              }
            />
            <input
              className={chars[char].id}
              type="radio"
              name={`${char}`}
              value="3"
              onClick={() =>
                handleCharacteristics(char, char[char].id, 3)
              }
            />
            <input
              className={chars[char].id}
              type="radio"
              name={`${char}`}
              value="4"
              onClick={() =>
                handleCharacteristics(char, chars[char].id, 4)
              }
            />
            <input
              className={chars[char].id}
              type="radio"
              name={`${char}`}
              value="5"
              onClick={() =>
                handleCharacteristics(char, `${chars[char].id}`, 5)
              }
            />
          </label>
          <br />
        </div>
      );
    }
    setCharsDiv(div);
  };

  return !showReviewModal ? (
    ''
  ) : (
    <div id="review-window">
      <div id="review-form">
        <h1>Write Your Review</h1>
        <h3>About the {productName}</h3>

        {/* This div will ask the customer to fill in a star rating */}
        <div id="rate-by-star">
          <fieldset>
            <legend>Overall Rating</legend>
            {rating >= 1 ? (
              <FontAwesomeIcon
                id="star-1"
                icon={solid('star')}
                onClick={() => {
                  handleClick(setRating, 1);
                }}
              />
            ) : (
              <FontAwesomeIcon
                id="star-1"
                icon={regular('star')}
                onClick={() => {
                  handleClick(setRating, 1);
                }}
              />
            )}
            {rating >= 2 ? (
              <FontAwesomeIcon
                id="star-2"
                icon={solid('star')}
                onClick={() => {
                  handleClick(setRating, 2);
                }}
              />
            ) : (
              <FontAwesomeIcon
                id="star-2"
                icon={regular('star')}
                onClick={() => {
                  handleClick(setRating, 2);
                }}
              />
            )}
            {rating >= 3 ? (
              <FontAwesomeIcon
                id="star-3"
                icon={solid('star')}
                onClick={() => {
                  handleClick(setRating, 3);
                }}
              />
            ) : (
              <FontAwesomeIcon
                id="star-3"
                icon={regular('star')}
                onClick={() => {
                  handleClick(setRating, 3);
                }}
              />
            )}
            {rating >= 4 ? (
              <FontAwesomeIcon
                id="star-4"
                icon={solid('star')}
                onClick={() => {
                  handleClick(setRating, 4);
                }}
              />
            ) : (
              <FontAwesomeIcon
                id="star-4"
                icon={regular('star')}
                onClick={() => {
                  handleClick(setRating, 4);
                }}
              />
            )}
            {rating === 5 ? (
              <FontAwesomeIcon
                id="star-5"
                icon={solid('star')}
                onClick={() => {
                  handleClick(setRating, 5);
                }}
              />
            ) : (
              <FontAwesomeIcon
                id="star-5"
                icon={regular('star')}
                onClick={() => {
                  handleClick(setRating, 5);
                }}
              />
            )}
          </fieldset>
        </div>

        {/* This div will ask the customer if they recommend the product*/}
        <div id="recommend">
          <fieldset>
            <legend>Do you recommend this product?</legend>
            <div>
              <label>
                <input
                  type="radio"
                  name="rec"
                  value="yes"
                  onClick={() => handleClick(setRecommend, true)}
                />{' '}
                Yes
              </label>
              <label>
                <input
                  type="radio"
                  name="rec"
                  value="no"
                  onClick={() => handleClick(setRecommend, false)}
                />{' '}
                No
              </label>
            </div>
          </fieldset>
        </div>
        {/* // This div asks the customer about different product characteristics */}
        <div id="characteristics-radios">
          <fieldset>
            <legend>Characteristics</legend>
            {charsDiv}
            <p> {`lowest(1)...........highest(5)`} </p>
          </fieldset>
        </div>

        {/* This div will alllow the user to enter a summary */}
        <div id="review-summary-input">
          <fieldset>
            <legend>Summary</legend>
            <div>
              <textarea
                maxLength="60"
                placeholder={'Example: Best purchse ever!'}
                rows="2"
                cols="35"
                onChange={(e) => handleChange(setSummary, e.target.value)}
              ></textarea>
            </div>
          </fieldset>
        </div>

        {/* This div will allow a user to enter a review body */}

        <div id="review-body-input">
          <fieldset>
            <legend>Review</legend>
            <div>
              <textarea
                minLength="50"
                maxLength="1000"
                placeholder={'Why did you like this product or not?'}
                rows="3"
                cols="70"
                required="required"
                wrap="hard"
                onChange={(e) => handleChange(setBody, e.target.value)}
              ></textarea>
            </div>
            <span>
              {body.length < 50
                ? `Minimum required characters left: ${50 - body.length}`
                : 'Minimum Reached'}
            </span>
          </fieldset>
        </div>

        {/* This div will allow the user to upload photos to the review */}

        <div>
          <fieldset>
            <legend>Upload your photos</legend>
            <input
              id="photo-input"
              type="file"
              accept="image/png, image/jpeg"
              multiple
              onChange={() => handlePhotos()}
            ></input>
            <div id="images"></div>
            {addPhotoDiv}
          </fieldset>
        </div>

        {/* This div will ask the user to enter their enter their name */}

        <div id="name-input">
          <fieldset>
            <legend>What is your Name?</legend>
            <input
              type="text"
              className="name-input"
              placeholder="Example: jackson11!"
              maxLength="60"
              size="35"
              onChange={(e) => handleChange(setName, e.target.value)}
            ></input>
            <br />
            <span>
              For privacy reasons, do not use your full name or email address
            </span>
          </fieldset>
        </div>

        {/* This div will ask the user to enter their email */}

        <div id="email-input">
          <fieldset>
            <legend>Your email</legend>
            <input
              type="email"
              className="email-input"
              placeholder="Example: jackson11@email.com"
              maxLength="60"
              size="35"
              onChange={(e) => handleChange(setEmail, e.target.value)}
            ></input>
            <br />
            <span>For authentication reasons, you will not be emailed</span>
          </fieldset>
          <br />
          <input type="submit" onClick={(e) => handleSubmit()}></input>
        </div>
      </div>
    </div>
  );
}

export default SubmitReview;
