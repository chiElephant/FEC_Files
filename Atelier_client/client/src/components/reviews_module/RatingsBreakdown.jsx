import React, { useState, useEffect } from 'react';
import ProductBreakdown from './ProductBreakdown.jsx';

import {
  getMetadata,
  metadata,
  ratings,
  characteristics,
} from './helper_functions/ratings_bd.jsx';

function RatingsBreakdown({ productId }) {
  const [characteristics, setCharacteristics] = useState(characteristics);
  const [ratingsGraphDiv, setRatingsGraphDiv] = useState(<div />);
  const [totalReviews, setTotalReviews] = useState(0);
  const [metadata, setMetadata] = useState(metadata);
  const [ratings, setRatings] = useState(ratings);
  const [percentRec, setPercentRec] = useState(0);
  const [avgRating, setAvgRating] = useState('');

  const setStates = {
    setCharacteristics,
    setRatingsGraphDiv,
    setTotalReviews,
    setPercentRec,
    setAvgRating,
    setMetadata,
    setRatings,
  };

  useEffect(() => {
    getMetadata(productId, setStates);
  }, [productId]);

  return (
    <div>
      <h3>Ratings Breakdown</h3>
      <div>{`Average Rating: ${avgRating}`}</div>
      <div>{`Total Reviews: ${totalReviews}`}</div>
      <br></br>
      <div id="rec-percent">
        {`${percentRec}% of reviewers recommend this product`}
      </div>
      <br></br>
      {ratingsGraphDiv}
      <br></br>
      <ProductBreakdown characteristics={characteristics} />
    </div>
  );
}

export default RatingsBreakdown;
