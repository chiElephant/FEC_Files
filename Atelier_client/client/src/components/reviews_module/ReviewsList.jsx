import React, { useState, onEffect } from 'react';
import handleSortType  from './helper_functions/review_list.jsx';
import Review from './Review.jsx';

function ReviewsList({ reviews, setSort, setType }) {
  const [sortType, setSortType] = useState('relevance');

  return (
    <div id="reviews-list">
      <h3>Reviews List</h3>
      <div>
        <label>Sort on: </label>
        <select
          id="sort-type"
          onChange={() => handleSortType(setSort, setType)}
        >
          <option value="relevant">Relevant</option>
          <option value="newest">Newest</option>
          <option value="helpful">Helpful</option>
        </select>
      </div>
      <ul id='reviews'>
        {reviews.map((review) => (
          <Review key={review.review_id} review={review} />
        ))}
      </ul>
    </div>
  );
}

export default ReviewsList;
