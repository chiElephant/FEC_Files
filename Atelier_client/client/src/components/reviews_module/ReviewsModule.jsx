import React, { useState, useEffect } from 'react';
import { helpers, initial } from './helper_functions/reviews_module.jsx';
import RatingsBreakdown from './RatingsBreakdown.jsx';
import SubmitReview from './SubmitReview.jsx';
import ReviewsList from './ReviewsList.jsx';

function ReviewsModule({ product_id, product_name }) {
  const [reviewsShown, setReviewsShown] = useState(initial.reviewModel);
  const [showReviewModal, setShowReviewModal] = useState(false);
  const [productName, setProductName] = useState(product_name);
  const [productId, setProductId] = useState(product_id);
  const [sortType, setSortType] = useState('relevance');
  const [countShown, setCountShown] = useState(2);
  const [reviews, setReviews] = useState(initial.reviewModel);

  // useEffect(() => {
  //   helpers.getReviews(productId, sortType, setReviews);
  // }, [productId, countShown, sortType]);

  // useEffect(() => {
  //   helpers.handleShown(reviews, countShown, setReviewsShown);
  // }, [reviews, countShown]);

  return (
    <div>
      <h2> Ratings Module </h2>
      <RatingsBreakdown productId={productId} />
      <ReviewsList
        reviews={reviewsShown}
        setSort={helpers.setSort}
        setType={setSortType}
      />
      <SubmitReview
        showReviewModal={showReviewModal}
        // submitReviewForm={helpers.submitReview}
        setShowReviewModal={setShowReviewModal}
        productName={productName}
        product_id={productId}
      />
      <div id="main-buttons">
        <button
          onClick={() =>
            helpers.handleClick(setShowReviewModal, !showReviewModal)
          }
        >
          Add Review
        </button>
        <button
          onClick={() => helpers.handleClick(setCountShown, countShown + 2)}
          disabled={countShown >= reviews.length}
        >
          More Reviews
        </button>
      </div>
    </div>
  );
}

export default ReviewsModule;
