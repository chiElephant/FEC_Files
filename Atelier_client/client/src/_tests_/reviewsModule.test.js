import renderer from 'react-test-renderer';
import ReviewsModule from '../components/reviews_module/ReviewsModule.jsx';
import ReviewsList from '../components/reviews_module/ReviewsList.jsx';
import Review from '../components/reviews_module/Review.jsx';
import ProductBreakdown from '../components/reviews_module/ProductBreakdown.jsx';
import RatingsBreakdown from '../components/reviews_module/RatingsBreakdown.jsx';
import SubmitReview from '../components/reviews_module/SubmitReview.jsx';

//Reviews Module
test('Reviews Module renders correctly', () => {
  const tree = renderer
    .create(<ReviewsModule />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


// //Ratings Breakdown
// test('Ratings Breakdown renders correctly', () => {
//   const tree = renderer
//     .create(<RatingsBreakdown />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

// //Add Reviews
// test('Submit Review renders correctly', () => {
//   const tree = renderer
//     .create(<SubmitReview />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });


