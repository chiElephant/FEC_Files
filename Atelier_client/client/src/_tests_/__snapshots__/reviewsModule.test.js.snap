// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings Breakdown renders correctly 1`] = `
<div
  id="ratings-breakdown"
>
  <h3>
    Ratings Breakdown
  </h3>
  <div>
    Average Rating: 
  </div>
  <div>
    Total Reviews: 
  </div>
  <br />
  <div
    id="rec-percent"
  >
    % of reviewers recommend this product
  </div>
  <br />
  <div />
  <br />
  <div>
    <h3>
      Product Breakdown
    </h3>
    <div
      id="char-div"
    />
  </div>
</div>
`;

exports[`Reviews Module renders correctly 1`] = `
<div>
  <h2>
     Ratings Module 
  </h2>
  <div
    id="ratings-breakdown"
  >
    <h3>
      Ratings Breakdown
    </h3>
    <div>
      Average Rating: 
    </div>
    <div>
      Total Reviews: 
    </div>
    <br />
    <div
      id="rec-percent"
    >
      % of reviewers recommend this product
    </div>
    <br />
    <div />
    <br />
    <div>
      <h3>
        Product Breakdown
      </h3>
      <div
        id="char-div"
      >
        <div />
      </div>
    </div>
  </div>
  <div
    id="reviews-list"
  >
    <h3>
      Reviews List
    </h3>
    <div>
      <label>
        Sort on: 
      </label>
      <select
        id="sort-type"
        onChange={[Function]}
      >
        <option
          value="relevant"
        >
          Relevant
        </option>
        <option
          value="newest"
        >
          Newest
        </option>
        <option
          value="helpful"
        >
          Helpful
        </option>
      </select>
    </div>
    <ul
      id="reviews"
    >
      <div
        id="review"
      >
        <h4>
          **********
        </h4>
        <div>
          Avgerage Rating: 
        </div>
        <div>
          Date Reviewed: 
        </div>
        <div>
          User: 
        </div>
        <h4 />
        <div />
        <div>
          <div />
        </div>
        <div />
        <div />
        <div>
          Helpfulness: 
        </div>
      </div>
    </ul>
  </div>
  <div
    id="main-buttons"
  >
    <button
      onClick={[Function]}
    >
      Add Review
    </button>
    <button
      disabled={true}
      onClick={[Function]}
    >
      More Reviews
    </button>
  </div>
</div>
`;

exports[`Submit Review renders correctly 1`] = `null`;
